daemon/Background.js

```diff
@@ -65,7 +65,8 @@ class Background extends Clutter.Actor {
     // Switch monitor side when the preview-on-right-side settings key changes.
     this._settings.connect('changed::preview-on-right-side', () => {
       if (this._previewMode) {
-        this.x = this._settings.get_boolean('preview-on-right-side') ? this.width : 0;
+        // Set x accounting monitor x as a starting point
+        this.x = this._settings.get_boolean('preview-on-right-side') ? this.width + Main.layoutManager.currentMonitor.x : Main.layoutManager.currentMonitor.x;
```

Comment:
Maybe this long line would be a bit more readable like this?

```suggestion
        this.x = Main.layoutManager.currentMonitor.x;
        if (this._settings.get_boolean('preview-on-right-side')) {
          this.x += this.width;
        }
```